### PR TITLE
Misc fixes, improvements and extensions

### PR DESCRIPTION
**Pull request by sirthias**
I'd like to propose a few changes/extensions that came out of my work on the [swave](http://swave.io) docs so far.

### CODE REVIEW

**Comment by ktoso** [comment]
Good idea with the base_url's, me likes 👍

**Comment by dwijnand** [comment]
Generally LGTM, but this further undermines Paradox's principle of the markdown being usable as source on GitHub, which I think is a shame.

**Comment by sirthias** [comment]
There is a tradeoff here, between keeping the links in the docs DRY and making them github friendly. I like that paradox gives me as a user the ability to choose my own sweet spot on this tradeoff continuum. Depending on how the dependent project chooses to manage itself the github linkability might be really important or merely nice to have.
I like that paradox isn't overly opinionated in this regard.

**Comment by dwijnand** [comment]
I see what you're saying.

I guess I highly value begin GitHub friendly, so I want to keep Paradox opinionated as such.

But if this helps adoption then ok.

File: core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala [hunk 1]
```diff
@@ -234,11 +234,15 @@ case class GitHubDirective(currentPath: String, variables: Map[String, String])
  *
  * Extracts snippets from source files into verbatim blocks.
  */
-case class SnipDirective(page: Page) extends LeafBlockDirective("snip") {
+case class SnipDirective(page: Page, variables: Map[String, String]) extends LeafBlockDirective("snip") {
+  private lazy val snippetBase =
+    new File(variables.getOrElse("snippet.base_dir", sys.error("Property `snippet.base_dir` is not defined")))
```
Comment (dwijnand) [inline core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala]: Won't this explode in the cases where you don't define it? (Same for image.base_url)

Comment (sirthias) [inline core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala]: Note: It's a `lazy val` though!

Comment (dwijnand) [inline core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala]: that's it, thanks :)

**Pull request merged**